### PR TITLE
Update docs for `ApproxTimeEvolution` to reference `TrotterProduct`

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Context:**
Once the new `TrotterProduct` functionality is added to Pennylane, the `ApproxTimeEvolution` class will be redundant. In future releases it will be deprecated and removed. For now we point to the new functionality in the doc string.

### CODE REVIEW

File: pennylane/templates/subroutines/approx_time_evolution.py [hunk 1]
```diff
@@ -24,6 +24,12 @@ class ApproxTimeEvolution(Operation):
     r"""Applies the Trotterized time-evolution operator for an arbitrary Hamiltonian, expressed in terms
     of Pauli gates.
 
+        .. note::
```
Comment (soranjh) [inline pennylane/templates/subroutines/approx_time_evolution.py:27]: Does it make any sense to mention this new feature in `qml.evolve` docs?

**Review by soranjh — APPROVED** [review]
Thanks @Jaybsoni.

File: pennylane/templates/subroutines/approx_time_evolution.py [hunk 2]
```diff
@@ -24,6 +24,12 @@ class ApproxTimeEvolution(Operation):
...
+
+            This operation behaves identically to :class:`~pennylane.TrotterProduct` and will be deprecated in
```
Comment (trbromley) [inline pennylane/templates/subroutines/approx_time_evolution.py:29]: Maybe "is a special case of"?

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 1 above)
Comment (trbromley) [inline pennylane/templates/subroutines/approx_time_evolution.py:27]: Is this too indented?

https://xanaduai-pennylane--4679.com.readthedocs.build/en/4679/code/api/pennylane.ApproxTimeEvolution.html

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 2 above)
Comment (trbromley) [reply pennylane/templates/subroutines/approx_time_evolution.py:29]: Are we merging this after TrotterProduct so we can confirm the links work?

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 1 above)
Comment (Jaybsoni) [reply pennylane/templates/subroutines/approx_time_evolution.py:27]: Nope, it should be independent of it.

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 2 above)
Comment (Jaybsoni) [reply pennylane/templates/subroutines/approx_time_evolution.py:29]: Yes!

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 1 above)
Comment (Jaybsoni) [reply pennylane/templates/subroutines/approx_time_evolution.py:27]: Ya it looks a little funny 😅

**Review by KetpuntoG — APPROVED** [review]

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 2 above)
Comment (Jaybsoni) [reply pennylane/templates/subroutines/approx_time_evolution.py:29]: Added

File: pennylane/templates/subroutines/approx_time_evolution.py (hunk 1 above)
Comment (Jaybsoni) [reply pennylane/templates/subroutines/approx_time_evolution.py:27]: Fixed

**Pull request merged**